### ISSUE TITLE
Fonts don’t show up properly

### ISSUE THREAD

**Issue by sophiebits** [mathiasbynens/math-tex#2]
Maybe I'm missing something, but at https://mathiasbynens.github.io/math-tex/dist/example.html it looks like the right fonts aren't being loaded, which causes the rendered math to look significantly worse. This happens because the katex.min.css stylesheet is inlined:

```
@font-face{font-family:KaTeX_Main;src:url(fonts/KaTeX_Main-Regular.eot);src:url(fonts/KaTeX_Main-Regular.eot?#iefix) format('embedded-opentype'),url(fonts/KaTeX_Main-Regular.woff) format('woff'),url(fonts/KaTeX_Main-Regular.ttf) format('truetype');font-weight:400;font-style:normal}
```

but the fonts aren't available at that path relative to the built math-tex.html (they live in bower_components). Not sure the best way to fix this as I'm not familiar with how web components are usually distributed.

**Comment by mathiasbynens** [comment]
The build script copies the fonts over to the `dist` directory which is also part of this repository (so that gh-pages can work): https://github.com/mathiasbynens/math-tex/tree/gh-pages/dist

https://mathiasbynens.github.io/math-tex/dist/example.html looks like this for me:

![](https://i.imgur.com/iSSS2c8.png)

What browser are you using? Could you post a screenshot?

**Comment by sophiebits** [comment]
Yes, I see the same thing as your screenshot. KaTeX bundles its own fonts but those fonts aren't being loaded properly. The formula should look like this:

![image](https://cloud.githubusercontent.com/assets/6820/4498488/f34ef164-4a78-11e4-93f1-bc35bcebe207.png)

If you edit the CSS rules on that page to use Courier as a fallback font

![image](https://cloud.githubusercontent.com/assets/6820/4498540/70141d32-4a79-11e4-9bc4-de068ce237e8.png)

then it may be more apparent that the fonts aren't being loaded:

![image](https://cloud.githubusercontent.com/assets/6820/4498547/77373b58-4a79-11e4-9d96-4b329d148f67.png)

Let me know if that makes sense.

**Comment by mathiasbynens** [comment]
Thanks for clarifying, @spicyj! Looking at the DevTools Network pane, it seems the fonts are not even requested. WTF?

**Comment by mathiasbynens** [comment]
Why does `@font-face` work as expected on https://mathiasbynens.github.io/html-imports-font-face/standalone.html but not on https://mathiasbynens.github.io/html-imports-font-face/example.html? ([repo link](https://github.com/mathiasbynens/html-imports-font-face))

`@font-face` doesn’t seem to work from within a `<template>` inside of an HTML Import, but I don’t know why. Does any spec mandate this, or is this an implementation bug that just happens to be present in both Chromium and Gecko? Or am I doing something very wrong here?

**Comment by mathiasbynens** [comment]
This seems to be [Chromium issue #336876](https://code.google.com/p/chromium/issues/detail?id=336876). Gecko seems to have [a similar problem](https://bugzilla.mozilla.org/show_bug.cgi?id=1077396). Thanks to @addyosmani for pinning it down!

**Comment by addyosmani** [comment]
My pleasure (and sorry that we haven't quite fixed this yet!)

**Comment by mathiasbynens** [comment]
A [workaround](https://twitter.com/addyosmani/status/518019995952226304) suggested by @addyosmani: include `<link rel=stylesheet href=css>` in the imported document outside of the `<template>`.

For this project, I’ve considered that before but decided against it for the first reason mentioned here: https://github.com/mathiasbynens/math-tex/issues/3#issuecomment-57652734

Also, in Addy’s example the parent document contains the CSS that applies the font to the document, so it’s not really a solution: it’s just moving the CSS from the web component to the outer document. :(
